packages/util-retry/src/StandardRetryStrategy.ts

```diff
@@ -1,47 +1,79 @@
+import { createDefaultRetryToken } from "./defaultRetryToken";
 
 /**
  * @public
  */
 export class StandardRetryStrategy implements RetryStrategyV2 {
   public readonly mode: string = RETRY_MODES.STANDARD;
-  private retryToken: StandardRetryToken;
+  private readonly capacityBucket = { availableCapacity: INITIAL_RETRY_TOKENS };
+  private readonly retryBackoffStrategy = getDefaultRetryBackoffStrategy();
   private readonly maxAttemptsProvider: Provider<number>;
 
   constructor(maxAttempts: number);
   constructor(maxAttemptsProvider: Provider<number>);
   constructor(private readonly maxAttempts: number | Provider<number>) {
-    this.retryToken = getDefaultRetryToken(INITIAL_RETRY_TOKENS, DEFAULT_RETRY_DELAY_BASE);
     this.maxAttemptsProvider = typeof maxAttempts === "function" ? maxAttempts : async () => maxAttempts;
   }
 
   public async acquireInitialRetryToken(retryTokenScope: string): Promise<StandardRetryToken> {
-    return this.retryToken;
+    return createDefaultRetryToken({
+      capacityBucket: this.capacityBucket,
+      retryDelay: DEFAULT_RETRY_DELAY_BASE,
+      retryCount: 0,
+    });
   }
 
   public async refreshRetryTokenForRetry(
-    tokenToRenew: StandardRetryToken,
+    token: StandardRetryToken,
     errorInfo: RetryErrorInfo
   ): Promise<StandardRetryToken> {
     const maxAttempts = await this.getMaxAttempts();
 
-    if (this.shouldRetry(tokenToRenew, errorInfo, maxAttempts)) {
-      tokenToRenew.getRetryTokenCount(errorInfo);
-      return tokenToRenew;
+    const retryCost = RETRY_COST;
+    const timeoutRetryCost = TIMEOUT_RETRY_COST;
+    const getCapacityCost = (errorType: RetryErrorType) => (errorType === "TRANSIENT" ? timeoutRetryCost : retryCost);
+
+    if (this.shouldRetry(token, errorInfo, maxAttempts)) {
+      const errorType = errorInfo.errorType;
+      const capacityCost = getCapacityCost(errorType);
+      const delayBase = errorType === "THROTTLING" ? THROTTLING_RETRY_DELAY_BASE : DEFAULT_RETRY_DELAY_BASE;
+      this.retryBackoffStrategy.setDelayBase(delayBase);
+      const delayFromErrorType = this.retryBackoffStrategy.computeNextBackoffDelay(token.getRetryCount());
+      let retryDelay: number;
+      if (errorInfo.retryAfterHint) {
+        const delayFromRetryAfterHint = errorInfo.retryAfterHint.getTime() - Date.now();
+        retryDelay = Math.max(delayFromRetryAfterHint || 0, delayFromErrorType);
+      } else {
+        retryDelay = delayFromErrorType;
+      }
+      this.capacityBucket.availableCapacity -= capacityCost;
+      return createDefaultRetryToken({
+        capacityBucket: this.capacityBucket,
+        retryDelay,
+        retryCount: token.getRetryCount() + 1,
+        lastRetryCost: capacityCost,
+      });
     }
+
     throw new Error("No retry token available");
   }
 
   public recordSuccess(token: StandardRetryToken): void {
-    this.retryToken.releaseRetryTokens(token.getLastRetryCost());
+    token.releaseRetryTokens(token.getLastRetryCost());
```

Comment:
Discussed offline: The tokens will be released in StandardRetryStrategy, and we're marking `releaseRetryTokens` as deprecated.